nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/test/java/org/apache/nifi/snmp/processors/SetSNMPTest.java

```diff
@@ -0,0 +1,159 @@
+        assertNotNull(failureFF);
+    }
+
+    private TestRunner getTestRunner(String port) {
+        SetSNMP processor = new SetSNMP();
+        TestRunner runner = TestRunners.newTestRunner(processor);
+        MockFlowFile ff = new MockFlowFile(123);
+        final Map<String, String> attributes = ff.getAttributes();
+        Map<String, String> newAttributes = new HashMap<>(attributes);
+        newAttributes.put("snmp$1.3.6.1.4.1.32437.1.5.1.4.2.0$4", testOIDValue);
+        ff.putAttributes(newAttributes);
+        runner.enqueue(ff);
+        runner.setProperty(GetSNMP.AGENT_HOST, "127.0.0.1");
+        runner.setProperty(GetSNMP.AGENT_PORT, port);
+        runner.setProperty(GetSNMP.SNMP_COMMUNITY, "public");
+        runner.setProperty(GetSNMP.SNMP_VERSION, "SNMPv1");
+        return runner;
+    }
+
+    private TestRunner getTestRunnerWithEmptyFlowFile(String port) {
+        SetSNMP processor = new SetSNMP();
+        TestRunner runner = TestRunners.newTestRunner(processor);
+        runner.enqueue(new MockFlowFile(123));
+        runner.setProperty(GetSNMP.AGENT_HOST, "127.0.0.1");
+        runner.setProperty(GetSNMP.AGENT_PORT, port);
+        runner.setProperty(GetSNMP.SNMP_COMMUNITY, "public");
+        runner.setProperty(GetSNMP.SNMP_VERSION, "SNMPv1");
+        return runner;
+    }
+
+    private TestRunner getTestRunnerWithInvalidHost() {
+        SetSNMP processor = new SetSNMP();
+        TestRunner runner = TestRunners.newTestRunner(processor);
+        MockFlowFile ff = new MockFlowFile(123);
+        final Map<String, String> attributes = ff.getAttributes();
+        Map<String, String> newAttributes = new HashMap<>(attributes);
+        newAttributes.put("snmp$1.3.6.1.4.1.32437.1.5.1.4.2.0$4", testOIDValue);
+        ff.putAttributes(newAttributes);
+        runner.enqueue(ff);
+        runner.setProperty(GetSNMP.AGENT_HOST, "127.0.0.2");
+        runner.setProperty(GetSNMP.AGENT_PORT, "1234");
+        runner.setProperty(GetSNMP.SNMP_COMMUNITY, "public");
+        runner.setProperty(GetSNMP.SNMP_VERSION, "SNMPv1");
+        return runner;
+    }
+
+    private TestRunner getTestRunnerWithInvalidOID() {
+        SetSNMP processor = new SetSNMP();
+        TestRunner runner = TestRunners.newTestRunner(processor);
+        MockFlowFile ff = new MockFlowFile(123);
+        final Map<String, String> attributes = ff.getAttributes();
+        Map<String, String> newAttributes = new HashMap<>(attributes);
+        newAttributes.put("snmp$1.3.6.1.4.1.32437.1.5.1.4.213.0$4", testOIDValue);
+        ff.putAttributes(newAttributes);
+        runner.enqueue(ff);
+        runner.setProperty(GetSNMP.AGENT_HOST, "127.0.0.1");
+        runner.setProperty(GetSNMP.AGENT_PORT, String.valueOf(snmpV1Agent.getPort()));
+        runner.setProperty(GetSNMP.SNMP_COMMUNITY, "public");
+        runner.setProperty(GetSNMP.SNMP_VERSION, "SNMPv1");
+        return runner;
```

Comment:
Like above: I see a lot of duplication it might worth to extract parts of the test methods